lib/metrics/metrics.go

```diff
@@ -21,48 +21,133 @@
 package metrics
 
 import (
+	"context"
+
 	"go.k6.io/k6/stats"
 )
 
-// TODO: refactor this, using non thread-safe global variables seems like a bad idea for various reasons...
+const (
+	VUsName               = "vus"
```

Comment:
This for example means that the cloud output will have to have access to the registry in order to make half the changes that are currently there which made .... a lot more code changes that were only needed for this specific case so I do prefer the constants to that.

Reply:
> This for example means that the cloud output will have to have access to the registry

Why? Wouldn't it just need access to the `BuiltInMetrics`, which it can easily get through https://github.com/k6io/k6/blob/7da10ac81128940e0828d9aca2e955f4e046c742/output/types.go#L100-L104

Reply:
> no good reason to not have the names of the metrics as constants

Not sure if it's a good reason, but my reason is that, when we have this metrics registry, there _shouldn't_ really be a need to compare metric names. And the fact that there is such a need is a bit of a code smell, something we probably need to refactor further in the future, potentially in another PR.

When every metric has a single object, you should be able to just compare these object pointers to each other, or use them as map indexes, and that would be enough (and more efficient than using strings). IMO metric names are an implementation detail, the majority of the k6 code shouldn't know them or care about them. The cloud output is a bit different, for now, because it repackages the samples in `http_req_li_all` and `http_iter_li_all`, but it should still be easily implementable by passing it the `BuiltInMetrics`.

What are the other places where metric names are used? As far as I can see, they are only required in the cloud output:
```
(git)-[metricsRegistry] % rg 'metrics\.[\w]+Name'
output/cloud/output_test.go
232:		Metric: metrics.VUsName,
650:		Metric: metrics.HTTPReqDurationName,

output/cloud/output.go
380:				metrics.DataSentName:     float64(sc.BytesWritten),
381:				metrics.DataReceivedName: float64(sc.BytesRead),
385:				values[metrics.IterationDurationName] = stats.D(sc.EndTime.Sub(sc.StartTime))
386:				values[metrics.IterationsName] = 1

output/cloud/data.go
119:	values[metrics.HTTPReqsName] = 1
120:	values[metrics.HTTPReqDurationName] = stats.D(trail.Duration)
121:	values[metrics.HTTPReqBlockedName] = stats.D(trail.Blocked)
122:	values[metrics.HTTPReqConnectingName] = stats.D(trail.Connecting)
123:	values[metrics.HTTPReqTLSHandshakingName] = stats.D(trail.TLSHandshaking)
124:	values[metrics.HTTPReqSendingName] = stats.D(trail.Sending)
125:	values[metrics.HTTPReqWaitingName] = stats.D(trail.Waiting)
126:	values[metrics.HTTPReqReceivingName] = stats.D(trail.Receiving)
128:		values[metrics.HTTPReqFailedName] = stats.B(trail.Failed.Bool)

output/cloud/data_test.go
53:				Metric: metrics.VUsName,
71:						metrics.DataSentName:          1234.5,
72:						metrics.DataReceivedName:      6789.1,
73:						metrics.IterationDurationName: stats.D(10 * time.Second),
```